src/object-store/src/test_util.rs

```diff
@@ -28,7 +31,34 @@ impl TempFolder {
         }
     }
 
-    pub async fn remove_all(&mut self) -> Result<()> {
+    pub async fn remove_all(&self) -> Result<()> {
         self.store.remove_all(&self.path).await
     }
 }
+
+/// Test s3 config from environment variables
+#[derive(Debug)]
+pub struct TestS3Config {
+    pub root: String,
+    pub access_key_id: String,
+    pub secret_access_key: String,
+    pub bucket: String,
+    pub region: Option<String>,
+}
+
+/// Returns s3 test config, return None if not found.
+pub fn s3_test_config() -> Option<TestS3Config> {
+    if let Ok(b) = env::var("GT_S3_BUCKET") {
+        if !b.is_empty() {
+            return Some(TestS3Config {
+                root: uuid::Uuid::new_v4().to_string(),
+                access_key_id: env::var("GT_S3_ACCESS_KEY_ID").unwrap(),
+                secret_access_key: env::var("GT_S3_ACCESS_KEY").unwrap(),
+                bucket: env::var("GT_S3_BUCKET").unwrap(),
+                region: Some(env::var("GT_S3_REGION").unwrap()),
```

Comment:
[nit]: maybe eat some sugar instead of just panic here when s3 env is not properly configured.
```suggestion
                access_key_id: env::var("GT_S3_ACCESS_KEY_ID").ok()?,
                secret_access_key: env::var("GT_S3_ACCESS_KEY").ok()?,
                bucket: env::var("GT_S3_BUCKET").ok()?,
                region: Some(env::var("GT_S3_REGION").ok()?),
```